File: src/main/java/com/lannstark/lec07/Lec07Main.java
package com.lannstark.lec07;

import org.jetbrains.annotations.NotNull;

import java.io.IOException;

public class Lec07Main {

  public static void main(String[] args) throws IOException {

  }

  private int parseIntOrThrow(@NotNull String str) {
    try {
      return Integer.parseInt(str);
    }catch (NumberFormatException e) {
      throw new IllegalArgumentException(String.format("주어진 %s는 숫자가 아닙니다."));
    }
  }

  private Integer parseIntOrThrowV2(@NotNull String str) {
    try {
      return Integer.parseInt(str);
    }catch (NumberFormatException e) {
      return null;
    }
  }
}

File: src/main/kotlin/com/lannstark/lec07/Lec07Main.kt
package com.lannstark.lec07

import java.lang.NumberFormatException


fun parseIntOrThrow(str: String): Int {
    try {
        return str.toInt()
    } catch (e: NumberFormatException) {
        throw IllegalArgumentException("주어진 ${str}는 숫자가 아닙니다.")
    }
}

fun parseIntOrThrowV2(str:String): Int? {
    return try { // 하나의 Expression 이기 때문에 return 가능
        str.toInt()
    } catch (e: NumberFormatException) {
        null
    }
}